BankDemo.sql

```diff
@@ -0,0 +1,421 @@
+)
+
+insert into Banks (Id, Name)
+	values ('BAPB', 'Belagroprombank'),
+			('AKBB', 'Belarusbank'),
+			('BLBB', 'Belinvestbank'),
+			('ALFA', 'Alpha-Bank'),
+			('POIS', 'Paritetbank'),
+			('PJCB', 'Priorbank')
+
+select * from Banks
+go
+
+create table Banks_Branches
+(
+	Id integer primary key,
+	Bank nvarchar(10) not null,
+	City nvarchar(6) not null,
+	Address nvarchar(100) not null,
+	foreign key (Bank) references Banks(Id),
+	foreign key (City) references Cities(Id),
+)
+
+insert into Banks_Branches(Id, Bank, City, Address)
+	values (1001, 'AKBB', 'MSK', 'Minsk, pr. Nezavisemosti, 91'),
+			(2001, 'AKBB', 'MGL', 'Mogilev, pr. Pushkinskiy, 34-41'),
+			(1045, 'BAPB', 'MSK', 'Minsk, pr. Pobediteley, 84'),
+			(4341, 'ALFA', 'VTBK', 'Vitebsk, ul. Cosmonavtov, 11'),
+			(6627, 'BLBB', 'GDN', 'Grodno, el. Mitskevicha, 3'),
+			(3441, 'POIS', 'GML', 'Gomel, pr. Lenina, 29A'),
+			(4567, 'PJCB', 'BST', 'Brest, b-r Shevchenko, 6/1')
+
+select * from Banks_Branches
+go
+
+create table Social_Statuses
+(
+	Id nvarchar(10) primary key,
+	Name nvarchar(100) not null,
+)
+
+insert into Social_Statuses (Id, Name)
+	values ('ST_EMP', 'State employee'),
+			('PENS', 'Pensioner'),
+			('DIS', 'Disabled person'),
+			('VET', 'Veteran'),
+			('LGFAM', 'Have a large family'),
+			('ORPH', 'Orphan')
+
+select * from Social_Statuses
+go
+
+create table Clients
+(
+	Id integer identity(1,1) primary key,
+	Surname nvarchar(100) not null,
+	Firstname nvarchar(30) not null,
+	Patronymic nvarchar(100),
+	Passport_number nvarchar(9) not null unique,
+	Phone_number nvarchar(13) not null,
+)
+
+insert into Clients (Surname, Firstname, Patronymic, Passport_number, Phone_number)
+	values ('Steblov', 'Alexander', 'Vasilievich', 'MP5682315', '80296724144'),
+			('Kureichik', 'Natalia', 'Georgievna', 'AB6281552', '+375446776635'),
+			('Burak', 'Tatiana', 'Nikolaevna', 'BM6788127', '+375295288241'),
+			('Vasiliev', 'Petr', 'Leonidovich', 'HB5689124', '80253565251'),
+			('Shibko', 'Anastasia', 'Alexandrovna', 'MC5689781', '80294556121'),
+			('Klukach', 'Pavel', 'Nikolaevich', 'MP4545215', '+375442234567'),
+			('Paramonov', 'Aleksei', 'Evgenievna', 'HB5625356', '80254544227')
+
+select * from Clients;
+go
+
+create table Clients_Social_Statuses
+(
+	Id integer identity(1,1) primary key,
+	Client_Id integer not null,
+	Status_Id nvarchar(10) not null,
+	foreign key (Client_Id) references Clients(Id),
+	foreign key (Status_Id) references Social_Statuses(Id),
+)
+
+insert into Clients_Social_Statuses(Client_Id, Status_Id)
+	values (1, 'PENS'),
+			(2, 'LGFAM'),
+			(3, 'PENS'),
+			(4, 'ORPH'),
+			(5, 'ST_EMP'),
+			(6, 'ST_EMP'),
+			(7, 'DIS'),
+			(7, 'VET')
+
+select * from Clients_Social_Statuses
+go
+
+create table Accounts
+(
+	Id nvarchar(20) primary key,
+	Client_Id integer not null,
+	Bank_Id nvarchar(10) not null,
+	Balance money not null,
+	foreign key (Client_Id) references Clients(Id),
+	foreign key (Bank_Id) references Banks(Id),
```

Comment:
It's some kind of a mess, it should be like:
Client can have multiple accounts
Bank can have multiple accounts
Account can have multiple cards
But one account can have only one bank and client
Read about normalization of database
![image](https://user-images.githubusercontent.com/33979910/202873546-d502be97-407f-4fe0-8b00-a73d568547e5.png)